### PR TITLE
Slight reorganization around top-of-file comments

### PR DESCRIPTION
**Pull request by fbartho**
I think this makes the code a little easier to follow.

I wasn't able to add an end-to-end integration test around "empty-statements-at-top-of-file" because you're right @IanVS, other layers prevent that from happening.

Instead I added a unit-test to assert the fatal-errors that we throw.

### CODE REVIEW

File: src/utils/get-comment-registry.ts [hunk 1]
```diff
@@ -243,7 +244,7 @@ export const getCommentRegistryFromImportDeclarations = ({
     /** Constructed Output Nodes */
     outputNodes: ImportDeclaration[];
 }) => {
-    if (outputNodes.length === 0 || !firstImport) {
+    if (outputNodes?.length === 0 || !firstImport) {
```
Comment (IanVS) [inline src/utils/get-comment-registry.ts:247]: When would `outputNodes` be undefined?  We have TS to guarantee it's an array just above, right?

File: src/utils/get-comment-registry.ts [hunk 2]
```diff
@@ -368,7 +369,12 @@ export function attachCommentsToOutputNodes(
     /** Original declaration, not the re-sorted output-node! */
     firstImport: ImportDeclaration,
 ) {
-    if (outputNodes.length === 0) {
+    if (!Array.isArray(commentEntriesFromRegistry)) {
```
Comment (IanVS) [inline src/utils/get-comment-registry.ts:372]: How would this happen without throwing a TypeScript compilation error?

File: src/utils/get-comment-registry.ts [hunk 3]
```diff
@@ -368,7 +369,12 @@ export function attachCommentsToOutputNodes(
...
+        throw new Error(
+            'Fatal Internal Error: Expected a list of commentEntriesFromRegistry',
+        );
+    }
+    if (outputNodes == null || outputNodes.length === 0) {
```
Comment (IanVS) [inline src/utils/get-comment-registry.ts:377]: Why check for a null `outputNodes`?

File: src/utils/get-comment-registry.ts (hunk 1 above)
Comment (fbartho) [reply src/utils/get-comment-registry.ts:247]: I concur that TypeScript would prevent that. 

I wasn’t sure how robust we wanted our `src/utils/*` to be against misuse.

I wrote a unit test that crashed this, so I made the check verify that it was non-null. All of my `Fatal Internal Error` checks could be removed if you wanted to do so. I found them helpful during the initial implementation phase.

File: src/utils/get-comment-registry.ts (hunk 2 above)
Comment (fbartho) [reply src/utils/get-comment-registry.ts:372]: I concur that TypeScript would prevent that. [Responded further here: https://github.com/IanVS/prettier-plugin-sort-imports/pull/84#discussion_r1194435397]

File: src/utils/get-comment-registry.ts (hunk 3 above)
Comment (fbartho) [reply src/utils/get-comment-registry.ts:377]: I concur that TypeScript would prevent that. [Responded further here: https://github.com/IanVS/prettier-plugin-sort-imports/pull/84#discussion_r1194435397]

**Review by IanVS — APPROVED** [review]
